src/scenario.js

```diff
@@ -21,26 +21,47 @@ const join = require('./tests/join');
 const sendMessage = require('./tests/send-message');
 const acceptInvite = require('./tests/accept-invite');
 const invite = require('./tests/invite');
-const receiveMessage = require('./tests/receive-message');
+const {
+    receiveMessage,
+    checkTimelineContains,
+    scrollToTimelineTop
+} = require('./tests/timeline');
 const createRoom = require('./tests/create-room');
 const changeRoomSettings = require('./tests/room-settings');
 const acceptServerNoticesInviteAndConsent = require('./tests/server-notices-consent');
-const getE2EDeviceFromSettings = require('./tests/e2e-device');
+const {enableLazyLoading, getE2EDeviceFromSettings} = require('./tests/settings');
 const verifyDeviceForUser = require("./tests/verify-device");
 
-module.exports = async function scenario(createSession) {
+module.exports = async function scenario(createSession, restCreator) {
     async function createUser(username) {
         const session = await createSession(username);
-        await signup(session, session.username, 'testtest');
+        await signup(session, session.username, 'testtest', session.hsUrl);
         await acceptServerNoticesInviteAndConsent(session);
         return session;
     }
 
     const alice = await createUser("alice");
     const bob = await createUser("bob");
+    const charlies = await createRestUsers(restCreator);
 
     await createDirectoryRoomAndTalk(alice, bob);
     await createE2ERoomAndTalk(alice, bob);
+    await aLazyLoadingTest(alice, bob, charlies);
+}
+
+function range(start, amount, step = 1) {
+    const r = [];
+    for (let i = 0; i < amount; ++i) {
+        r.push(start + (i * step));
+    }
+    return r;
+}
```

Comment:
ftr lodash also has this, but maybe not worth pulling it in for the sake of a 5 line function

Reply:
:+1: next time a need a lodash method I'll import it